include/deal.II/grid/grid_generator.h

```diff
@@ -2424,6 +2424,34 @@ namespace GridGenerator
                                     Triangulation<1, spacedim>       &out_tria);
 #endif
 
+
+  /**
+   * Perform an Alfeld split (also called barycentric refinement) of a simplex
+   * mesh.
+   *
+   * @note Currently only implemented for @p dim = 2.
+
+   * This function takes a simplex mesh (@p in_tria) and subdivides each
+   * triangle into three triangles with a single new vertex (the barycenter). In
+   * the process, the simplex mesh is flattened (no hierarchy is kept).
```

Comment:
What happens if you had hanging nodes? Do you end up with a mesh with an interior boundary? 

Instead of going back and forth between the word "simplex" and "triangle", I would settle on one.